test/new-e2e/tests/otel/utils/pipelines_utils.go

```diff
@@ -499,11 +500,48 @@ func getLoadBalancingSpans(t assert.TestingT, traces []*aggregator.TracePayload)
 			}
 		}
 	}
-	assert.Equal(t, 12, spans)
+	require.Equal(t, 12, spans)
 	return spanMap
 }
 
-func getLoadBalancingMetrics(t assert.TestingT, metrics []*aggregator.MetricSeries) map[string][]map[string]string {
+func getLoadBalancingLogs(c require.TestingT, s OTelTestSuite, service string) {
+	logs1, err := s.Env().FakeIntake.Client().FilterLogs(service, fakeintake.WithMessageContaining(log1Body))
+	require.NoError(c, err)
+	require.NotEmpty(c, logs1)
+	log1 := logs1[0]
+	log1Tags := getTagMapFromSlice(c, log1.Tags)
+	attrs := make(map[string]interface{})
+	err = json.Unmarshal([]byte(log1.Message), &attrs)
+	require.NoError(c, err)
+	for k, v := range attrs {
+		log1Tags[k] = fmt.Sprint(v)
+	}
+
+	logs2, err := s.Env().FakeIntake.Client().FilterLogs(service, fakeintake.WithMessageContaining(log2Body))
```

Comment:
QQ: log2Body is a sub-string of log1Body, so if you filter on log2Body would it be possible to still get log1 rather than log2?

Reply:
Yup, I think that would be the case. Good catch.

Reply:
True, just updated to use `WithMessageMatching` instead